TypeCobol/Compiler/Diagnostics/Cobol2002Checker.cs

```diff
@@ -93,13 +93,28 @@ public static void CheckTypeDefinition(TypeDefinition typeDefinition)
                 string message = "TYPEDEF \'" + typeDefinition.Name + "\' has no description.";
                 DiagnosticUtils.AddError(typeDefinition, message, MessageCode.SemanticTCErrorInParser);
             }
+
+            if (typeDefinition.IsInsideCopy())
```

Comment:
It's ok but for performance reason, this test should be called only during the CodeElement phase.
Method CheckTypeDefinition(TypeDefinition typeDefinition) is called on crossCheck phase, meaning that this test apply to all TypeDefinition.

If this test is moved to CodeElement phase, it'll only be called if the DataTypeDescriptionEntry (the CodeElement associated to the TypeDefinition) need to be refreshed.

So can you can create another static method in class TypeDefinitionChecker. This new method can be called  by CobolCodeElementBuilder.

Reply:
It was planned to include this check in TypeDefinitionEntryChecker.CheckTypedef but the IsInsideCopy method always returns false in this context because no tokens have been consumed yet. The location is not available at this phase.

Reply:
Ok right I forgot about that.